### ISSUE TITLE
Irrlicht does not use fullscreen resolution over VGA

### ISSUE THREAD

**Issue by emorrp1** [minetest/minetest#7689]
##### Issue type
 - Bug report
 
##### Minetest version
```
Minetest 0.4.17.1 (Linux)
Using Irrlicht 1.8.4
Build info: VER=0.4.17.1 BUILD_TYPE=None RUN_IN_PLACE=0 USE_GETTEXT=1 USE_SOUND=1 USE_CURL=1 USE_FREETYPE=1 USE_LUAJIT=1 STATIC_SHAREDIR="/usr/share/games/minetest"
```

##### OS / Hardware
Operating system: Debian Stretch
CPU: AMD A8-5600K

GPU model: Radeon HD 7560D
OpenGL version: Mesa 13.0.6

##### Summary
With my current monitor the maximum resolution is not autodetected, so I
have configured xrandr to add a new mode and this works perfectly for
lightdm/xfce.

Unfortunately, minetest somehow does not find this configuration so
always shows the titlebar unless I drop the resolution to at most
848x480, which is then really blurry. Additionally, the desktop
resolution is not reverted to the previous setting when quitting.

##### Steps to reproduce
1. Add real monitor resolution to xrandr [source](https://selivan.github.io/2017/08/16/lightdm-add-custom-display-resolution.html):
    mode=$( cvt 1600 900 60 | grep -v '^#' | cut -d' ' -f3- )
    xrandr --newmode "1600x900" $mode
    xrandr --addmode VGA-0 "1600x900"
    xrandr --output VGA-0 --mode "1600x900"
2. Configure graphics in minetest.conf:
    fullscreen = true
    screenW = 1600
    screenH = 900
3. Launch minetest:
    2018-07-30 11:46:38: WARNING[Main]: Irrlicht: Could not find specified video mode, running windowed.
4. It is not fullscreen and the screenH has been changed to 871.

**Comment by ClobberXD** [comment]
Not sure if that's the issue, as Minetest recognizes my VGA display and its resolution (1600x900) correctly, and fullscreen mode works as intended.

**Comment by emorrp1** [comment]
And did xrandr auto-detect your resolution? Because I can use any of the normal modes shown in `xrandr -q` in fullscreen, but not the higher resolution manually added to xrandr.

**Comment by ClobberXD** [comment]
My display is running at its maximum resolution; it wasn't manually added though.

**Comment by emorrp1** [comment]
In case it helps, here's the current output, the first 4 were detected and work in fullscreen. So I don't know if minetest is doing it's own display querying or something? All 5 work in XFCE display settings.

emorrp1@olorin:~$ xrandr -q
Screen 0: minimum 320 x 200, current 1600 x 900, maximum 16384 x 16384
HDMI-0 disconnected (normal left inverted right x axis y axis)
VGA-0 connected 1600x900+0+0 (normal left inverted right x axis y axis) 0mm x 0mm
   1024x768      60.00  
   800x600       60.32    56.25  
   848x480       60.00  
   640x480       59.94  
   1600x900      59.95* 
DVI-0 disconnected (normal left inverted right x axis y axis)

**Comment by Calinou** [comment]
Try using a borderless window for fullscreen instead of exclusive fullscreen (which is pretty bad due to Irrlicht on Linux; you won't be able to alt-tab out of Minetest when in fullscreen). Your window manager should have a way to do this (<kbd>Alt + F11</kbd> is bound by default for this on Xfwm).

**Comment by rubenwardy** [comment]
As calinou said, use borderless full screen instead. Minetest should do this too - I believe there's an issue for it

**Comment by emorrp1** [comment]
Thank you for the workaround, that seems to work (and doesn't just "stretch" the resolution).

I read the other fullscreen setting & shortcut related bugs first and initially didn't consider this to be a dupe of borderless. Having re-read how difficult true fullscreen is with multiple monitors etc., I do think this issue would be best solved by implementing #4921 (though as commented, it's sad that there isn't a per-platform fullscreen standard).
